code/Model/Indexer/Algolia.php

```diff
@@ -254,11 +254,18 @@ protected function _registerCatalogProductEvent(Mage_Index_Model_Event $event)
             case Mage_Index_Model_Event::TYPE_SAVE:
                 /** @var $product Mage_Catalog_Model_Product */
                 $product = $event->getDataObject();
+
+                //Reload required attributes instead of product for performance
+                $store = $product->getStore();
+                $productId = $product->getId();
+                $visibility = $product->getResource()->getAttributeRawValue($productId, 'visibility', $store);
```

Comment:
@raminv80 Probably worth adding a check to see if these attributes are already populated before loading them again. E.g. `$product->getVisability() !== false`. Also, can you add an explanation of why the attributes may be set to false (caused by product save event).